File: corpus/typescript/fixture_17.ts
type CartItem17 = { price: number; qty?: number };
export function calculateDiscount17(cartItems: CartItem17[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 170 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook17 {
  lookupPrice(sku: string): number { return sku.length + 17; }
}

File: corpus/go/fixture_17.go
package fixture17

type PriceBook17 struct {
    Multiplier float64
}

func CalculateDiscount17(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 170.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}
